src/receiver.mjs

```diff
@@ -0,0 +1,57 @@
+
+import AudioBackend from './backend/index.mjs'
+import receiveProcessor from './processors/receive.mjs'
+
+const getUserAudio = async ( ) =>
+    navigator.mediaDevices.getUserMedia( {
+        audio: {
+            echoCancellation: false
+        }
+    } )
+
+export class Receiver {
+    constructor( profileKey ) {
+        console.log( 'receiver initialized with profile', profileKey )
+        this.audioNode = null
+        this.backend = new AudioBackend( profileKey )
+    }
+
+    async start( ) {
+        this.audioNode = await this.backend.createAudioNode( receiveProcessor )
+        return this.resume( )
+    }
+
+    on( event, fn ) {
+        return this.backend.on( event, fn )
+    }
+
+    off( event, fn ) {
+        return this.backend.off( event, fn )
+    }
+
+    pause( ) {
+        this.audioNode.disconnect( )
+        this.audioStream.getAudioTracks( )
+            .map( track => track.stop( ) )
+    }
+
+    async resume( ) {
+        const context = this.backend.getContext( )
+
+        this.audioStream = await getUserAudio( )
```

Comment:
Taking in an audio stream was one of the few absolutes needed by jitsi, so jitsi (or the consuming app) could provide audio input or output from a specified device instead of letting the browser pick.

Reply:
aye, this was just a preliminary MR to test in VOD